File: compare-version-numbers/v2.py
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        v1 = list(map(int, version1.split(".")))
        v2 = list(map(int, version2.split(".")))

        for i in range(max(len(v1), len(v2))):
            a = v1[i : i + 1] or [0]
            b = v2[i : i + 1] or [0]
            place_diff = a[0] - b[0]
            if place_diff > 0:
                return 1
            elif place_diff < 0:
                return -1

        return 0


# Runtime: 51 ms, faster than 28.81% of Python3 online submissions for Compare Version Numbers.
# Memory Usage: 13.9 MB, less than 68.52% of Python3 online submissions for Compare Version Numbers.

File: partitioning-into-minimum-number-of-deci-binary-numbers/v2.go
package v2

func minPartitions(n string) int {
	i0 := int('0')
	var digit, highestDigit int
	for _, r := range n {
		digit = int(r) - i0
		if digit > highestDigit {
			highestDigit = digit
		}
	}

	return highestDigit
}

// Runtime: 10 ms, faster than 87.72% of Go online submissions for Partitioning Into Minimum Number Of Deci-Binary Numbers.
// Memory Usage: 6.4 MB, less than 94.74% of Go online submissions for Partitioning Into Minimum Number Of Deci-Binary Numbers.
